pdr_backend/analytics/predictoor_dashboard/dash_components/callbacks.py

```diff
@@ -138,14 +139,42 @@ def update_predictoors_table_on_search(search_value, predictoors_data):
         [
             Input("search-input-Feeds", "value"),
             Input("feeds-data", "data"),
+            Input("toggle-switch-predictoor-feeds", "value"),
+            Input("predictoors_table", "selected_rows"),
         ],
+        State("predictoors_table", "data"),
+        State("data-folder", "data"),
     )
-    def update_feeds_table_on_search(search_value, feeds_data):
-        if not search_value:
-            return feeds_data
+    def update_feeds_table_on_search(
+        search_value,
+        feeds_data,
+        predictoor_feeds_only,
+        predictoors_table_selected_rows,
+        predictoors_data,
+        lake_dir,
+    ):
+        # Extract selected predictoor addresses
+        predictoors_addrs = [
+            predictoors_data[i]["user"] for i in predictoors_table_selected_rows
+        ]
 
         # filter feeds by pair address
-        filtered_data = filter_objects_by_field(feeds_data, "pair", search_value)
+        filtered_data = (
```

Comment:
so why don't we integrate this in the query as well?
I know we discussed it. When searching for predictoors/feeds based on user input, we are still filtering from a list of objects instead of hitting the DB (with a LIKE query, no big deal). The current implementation is slow and should be addressed soon.

I asked you on Slack wrt to the functionality described here. Especially so, if we are hitting the database for feed ids based on predictoors, let us have not an id query, but a full feed query which also selects based on search value. This would take care of the feeds query.

We can adapt the predictoors part from a list to a db hit later, if outside the scope here.

Reply:
basically what we do right now is slow because of the following reasons:
- we retrieve the full feed list at app startup, and load it to the state
- in this function, we filter our list based on a keyword -> call it list A
- we hit the database **for another list!**: a list of feed ids -> call it list B
- we filter list A based on list B

So we have the startup database hit, the ids database hit and then the filter costs us O(len(A)) * O(len(B)), if you look at the list comprehension, especially since B is not ordered/hashed, so the lookup for "obj["contract"] in feed_ids" is not optimised: https://stackoverflow.com/questions/13884177/complexity-of-in-operator-in-python

Instead, what we can do:
- one database hit here, to retrieve all feed information based on two inputs: the selected predictoors and the pair search value input
- at startup, we can use the same query, without the LIKE filter

Reply:
db query to be addressed later